inc/fqdnlabel.class.php

```diff
@@ -151,25 +151,26 @@ static function getIDsByLabelAndFQDNID($label, $fqdns_id, $wildcard_search = fal
          if ($count == 0) {
             $label = '%'.$label.'%';
          }
-         $relation = "LIKE '$label'";
+         $relation = ['LIKE',  $label];
       } else {
-         $relation = "= '$label'";
+         $relation = $label;
       }
 
       $IDs = [];
       foreach (['NetworkName'  => 'glpi_networknames',
                      'NetworkAlias' => 'glpi_networkaliases'] as $class => $table) {
-         $query = "SELECT `id`
-                   FROM `$table`
-                   WHERE `name` $relation ";
-
-         if ((is_array($fqdns_id)) && (count($fqdns_id) > 0)) {
-            $query .= " AND `fqdns_id` IN ('". implode('\', \'', $fqdns_id). "')";
-         } else if ((is_int($fqdns_id)) && ($fqdns_id > 0)) {
-            $query .= " AND `fqdns_id` = '$fqdns_id'";
+         $criteria = [
+            'SELECT' => 'id',
+            'FROM'   => $table,
+            'WHERE'  => ['name' => $relation]
+         ];
+
+         if (is_array($fqdns_id) && count($fqdns_id) > 0 ||
+            is_int($fqdns_id) && $fqdns_id > 0) {
```

Comment:
CS note, i prefer writing the `||` part on the second line (i missed it in my first reading)

Reply:
I totally agree :)